src/producer.rs

```diff
@@ -557,8 +557,13 @@ impl<Exe: Executor> TopicProducer<Exe> {
         let (tx, rx) = oneshot::channel();
         match self.batch.as_ref() {
             None => {
-                let receipt = self.send_compress(message).await?;
-                let _ = tx.send(Ok(receipt));
+                let fut = self.send_compress(message).await?;
+                self.client
+                    .executor
+                    .spawn(Box::pin(async move {
+                        let _ = tx.send(fut.await);
```

Comment:
We put the request to a multi-thread executor. Is it possible to disorder the message?

Reply:
cc @cirias

Reply:
@mattisonchao Thank you for reviewing this!

The order of the messages should remain the same as the order of the `send`. Since each message is pushed to one channel underneath. https://github.com/cirias/pulsar-rs/blob/feature/non-blocking-send/src/connection.rs#L664